include/ascFile.py

```diff
@@ -89,3 +89,17 @@ def write_asc(ascheader, ascdata, filename):
             row = ' '.join(row)
             ascfile.write(row)
             ascfile.write('\n')
+
+
+# getting data ready for binning
+def bin_asc(filename):
+
+    # data should be 1-dimensional array, python list or iterable
+
+    with open(filename) as ascfile:
+        myArray = np.loadtxt(ascfile, skiprows=6)
+
+    array_1d = myArray.flatten()
+    return array_1d
```

Comment:
I see where you are going with this, but this really should be in something like `calibrationLib` instead. Also, still not sure why you need to use `loadtxt` to get the data.

Reply:
As I am using numpy, so I thought it would be better to convert it into numpy array and start processing the file. Let me know if doesn't look good.

Reply:
That's not unreasonable, but besides the `flatten` function where all are you using NumPy?

Reply:
For now its used only for `flatten`

Reply:
@PakhiAgarwal You can use an expression like this to do the same operation then: `array = list(i for j in matrix for i in j)`

Reply:
Yeah. We can use this expression, but I think using numpy will have same effect; the complexity of both is comparable. 
Do we have to avoid using built in libraries?

Reply:
@PakhiAgarwal That's a good question and to keep things brief, in addition to the computational complexity, its also important to keep in mind the broader aspects of a project. A general rule of thumb is to avoid importing additional dependencies unless there is a compelling need for them since the dependencies place additional burdens on a project (e.g., make sure they are documented, installed, upgraded, etc.). There was [a case a couple years ago](https://arstechnica.com/information-technology/2016/03/rage-quit-coder-unpublished-17-lines-of-javascript-and-broke-the-internet/) where the removal of a JavaScript dependency caused cascading failures for awhile. Since this would be adding a dependency to do something that we can do ourselves with a single line of code, it doesn't make much sense to add that dependency. Conversely, if we were making heavy use of NumPy throughout it would make sense to use the `flatten` function for consistency and readablity.

Reply:
Sounds reasonable.